test/05_deploy_crs_test.go

```
@@ -689,6 +704,52 @@ func TestDeployment_WaitForControlPlane(t *testing.T) {
 			}
 		}
 
+		// Stall detection: check if meaningful progress was made since last iteration
+		if stallEnabled {
+			// Build current state snapshot
+			currentMPReplicas := 0
+			if len(status.MachinePools) > 0 {
+				currentMPReplicas = status.MachinePools[0].ReadyReplicas
+			}
+			infraReady := 0
```

golangci-lint (2.11.4)

[error] assigned to infraReady, but reassigned without using the value

(wastedassign)